File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    hashmap := make(map[int]int)
    for i, num := range nums {
        complement := target - num
        if idx, found := hashmap[complement]; found {
            return []int{idx, i}
        }
        hashmap[num] = i
    }
    return nil 
}

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int complement, i = 0;
        for (int num : nums) {
            complement = target - num;
            if (map.containsKey(complement)) {
                return new int[]{map.get(complement), i};
            }
            map.put(num, i);
            i++;
        }
        throw new IllegalArgumentException();
    }
}